src/libtrx/game/objects/general/drawbridge.c

```diff
@@ -111,4 +127,87 @@ static void M_Setup(OBJECT *const obj)
     obj->floor_height_func = M_GetFloorHeight;
 }
 
+void M_DropStack(const ITEM *const item)
+{
+    VECTOR *pos = Vector_Create(sizeof(XYZ_32));
+    Drawbridge_GetSectorPositions(item, pos);
+    for (int32_t i = 0; i < pos->count; i++) {
+        MovableBlock_DropStack(
+            item->pos.y, *(const XYZ_32 *)Vector_Get(pos, i), item->room_num);
+    }
+    Vector_Free(pos);
+    pos = nullptr;
+}
+
+static int32_t floor_div(int32_t v, int32_t d)
+{
+    return (v >= 0) ? v / d : -((-v + d - 1) / d);
+}
+
+BOUNDS_16 M_RotateBounds(const BOUNDS_16 bounds, int16_t rot_y)
+{
+    BOUNDS_16 rot_bounds = { 0 };
+
+    switch (rot_y) {
+    case 0:
+    default:
+        rot_bounds = bounds;
+        break;
+    case DEG_90:
+        rot_bounds.min.x = bounds.min.z;
+        rot_bounds.max.x = bounds.max.z;
+        rot_bounds.min.z = -bounds.max.x;
+        rot_bounds.max.z = -bounds.min.x;
+        break;
+    case -DEG_180:
+        rot_bounds.min.x = -bounds.max.x;
+        rot_bounds.max.x = -bounds.min.x;
+        rot_bounds.min.z = -bounds.max.z;
+        rot_bounds.max.z = -bounds.min.z;
+        break;
+    case -DEG_90:
+        rot_bounds.min.x = -bounds.max.z;
+        rot_bounds.max.x = -bounds.min.z;
+        rot_bounds.min.z = bounds.min.x;
+        rot_bounds.max.z = bounds.max.x;
+        break;
+    }
+    return rot_bounds;
+}
+
+void Drawbridge_GetSectorPositions(const ITEM *const item, VECTOR *sector_pos)
+{
+    const OBJECT *const obj = Object_Get(item->object_id);
+    const ANIM_FRAME *const frame =
+        Object_GetAnim(obj, DRAWBRIDGE_ANIM_CLOSED)->frame_ptr;
+    const BOUNDS_16 rot_bounds = M_RotateBounds(frame->bounds, item->rot.y);
+
+    const int32_t x0 = item->pos.x + rot_bounds.min.x;
+    const int32_t x1 = item->pos.x + rot_bounds.max.x - 1; // inclusive
+    const int32_t z0 = item->pos.z + rot_bounds.min.z;
+    const int32_t z1 = item->pos.z + rot_bounds.max.z - 1;
+
+    const int32_t sx0 = floor_div(x0, WALL_L);
+    const int32_t sx1 = floor_div(x1, WALL_L);
+    const int32_t sz0 = floor_div(z0, WALL_L);
+    const int32_t sz1 = floor_div(z1, WALL_L);
+    // Sector of the drawbridge original position.
+    const int32_t sx_orig = floor_div(item->pos.x, WALL_L);
+    const int32_t sz_orig = floor_div(item->pos.z, WALL_L);
+
+    for (int32_t sx = sx0; sx <= sx1; ++sx) {
+        for (int32_t sz = sz0; sz <= sz1; ++sz) {
+            if (sx == sx_orig && sz == sz_orig)
+                // Skip the original sector since it's WALL_L away.
+                continue;
+
+            XYZ_32 pos = { .x = sx * WALL_L + WALL_L / 2,
+                           .y = item->pos.y,
+                           .z = sz * WALL_L + WALL_L / 2 };
```

Comment:
Would be cool to see this reformatted as
```c
            const XYZ_32 pos = {
                .x = sx * WALL_L + WALL_L / 2,
                .y = item->pos.y,
                .z = sz * WALL_L + WALL_L / 2,
            };
```
(trailing comma should trigger a reindent)